src/unk_02092BE8.c

```diff
@@ -0,0 +1,313 @@
+    Unk_PokegearSTRUCT_14 Unk_struct;
+    SCRIPT_STATE *state = SavArray_Flags_get(sys->savedata);
+    FLYPOINTS_SAVE *points = Save_FlyPoints_get(sys->savedata); 
+    Location *warpPtr = FlyPoints_GetSpecialSpawnWarpPtr(points); 
+    Location *PosPtr = FlyPoints_GetPosition(points); 
+    UnkFsysSub_114* Unk_V1 = sub_02092DEC(sys); 
+    ptr->data = sys->savedata;
+    ptr->mapID = PosPtr->mapId;
+    if (MapHeader_MapIsOnMainMatrix(ptr->mapID)) { 
+        ptr->x = GetPlayerXCoord(sys->playerAvatar);
+        ptr->y = GetPlayerYCoord(sys->playerAvatar);
+        ptr->matrixXCoord = ptr->x / 32;
+        ptr->matrixYCoord = ptr->y / 32;
+        ptr->mapHeader = ptr->mapID;
+    } else {
+        ptr->x = warpPtr->x;
+        ptr->y = warpPtr->z;
+        MapHeader_GetWorldMapCoords(ptr->mapID, &(Unk_struct.x), &(Unk_struct.y));
+        MAPMATRIX* matrix = MapMatrix_New();
+        MapMatrix_Load(MAP_T20, matrix);
+        if (Unk_struct.x == 0 && Unk_struct.y == 0) {
+            ptr->matrixXCoord = ptr->x / 32;
+            ptr->matrixYCoord = ptr->y /32;
+        } else {
+            ptr->matrixXCoord = Unk_struct.x;
+            ptr->matrixYCoord = Unk_struct.y;
+        }
+        ptr->mapHeader = MapMatrix_GetMapHeader(matrix, ptr->matrixXCoord, ptr->matrixYCoord);
+        MapMatrix_Free(matrix);
+    }
+
+    ptr->playerGender = PlayerProfile_GetTrainerGender(Sav2_PlayerData_GetProfileAddr(Fsys_GetSaveDataPtr(sys)));
+    ptr->unkPtr28 = &sys->unk_10C;
+    ptr->unkFsysVar01 = sub_0203DF3C(sys);
+    ptr->mapMusicID = Fsys_GetSurfOverriddenMusicId(sys, ptr->mapID);
+    
+    if (Unkbool != 0) {
+        return;
+    }
+    if (sub_02092E08(Unk_V1) != 0) {
+        ptr->unkVar04 = ov02_02251EE8(Unk_V1, &(Unk_struct.unkVar0c));
+        if (ptr->unkVar04 >= 0x4b) {
+            ptr->unkBool00 = 0;
+            return;
+        } else {
+            ptr->unkVar06 = Unk_struct.unkVar10;
+            ptr->unkVar05 = Unk_struct.unkVar0f;
+            if (ptr->unkVar05 == 3) {
+                sub_0202F050(SaveData_GetMomsSavingsAddr(sys->savedata), Unk_struct.unkVar0e);
+            }
+            else if (ptr->unkVar05 == 0) {
+                sub_0202AB18(Sav2_Misc_get(sys->savedata), Unk_struct.unkVar0c, Unk_struct.unkVar0d, ptr->unkVar04);
+            }
+            sub_02092F64(Unk_V1);
+            ptr->unkBool00 = 1;
+            return;
+    }
+    } else {
+        ptr->unkBool00 = 0;
+        return;
+    }
+}
+
+void sub_02092D80(FieldSystem* sys, Unk_PokegearSTRUCT_2C* ptr){
+    sub_02092BE8(sys, ptr, 0);
+}
+
+void sub_02092D8C(FieldSystem* sys, Unk_PokegearSTRUCT_2C* ptr){
+    sub_02092BE8(sys, ptr, 1);
+}
+
+UnkFsysSub_114* sub_02092D98(HeapID id, FieldSystem* sys){ 
+    UnkFsysSub_114 * ptr = AllocFromHeap(id, sizeof(UnkFsysSub_114));
+    MI_CpuFill8(ptr, 0, sizeof(UnkFsysSub_114));
+    sub_02092F64(ptr);
+    ptr->unk_varC = 0xa;
+    ptr->unk_var10 = 0x1e;
+    ptr->unk_var12 = 0x32;
```

Comment:
unless these values are bitfields, since they're relatively small, they should be expressed as decimal rather than hex